docstr_coverage/cli.py

```diff
@@ -100,6 +100,56 @@ def parse_ignore_names_file(ignore_names_file: str) -> tuple:
     return ignore_names
 
 
+def parse_ignore_patterns_from_dict(ignore_patterns_dict) -> tuple:
+    """
+    Parse dictionary containing (file_name_pattern, exclude_patterns) key value pairs
+    to return an output consistent with ignore patterns parsed by `parse_ignore_names_file`.
+
+    Note: To align the workflow with `parse_ignore_names_file`,
+    we check that the passed values have indeed type string,
+    but we do not yet check if they are valid regular expressions.
+
+    Parameters
+    ----------
+    ignore_patterns_dict: Dict
+        A dict where each key is a string and each value is a string or a nonempty list of strings.
+
+    Returns
+    -------
+    Tuple
+        Tuple of iterables of string,
+        with equal meaning as the output from `parse_ignore_names_file` """
+
+    def _assert_valid_key_value(key, value):
```

Comment:
I think we should rename the inner function's `key`/`value` parameters to `k`/`v`, since the outer scope already defines variables with these names further down

Reply:
Yep, you're absolutely right. I'll change that!